compiler/src/main/java/com/squareup/anvil/compiler/codegen/reference/AnnotationReferenceIr.kt

```diff
@@ -0,0 +1,80 @@
+package com.squareup.anvil.compiler.codegen.reference
+
+import com.squareup.anvil.compiler.api.AnvilCompilationException
+import com.squareup.anvil.compiler.internal.reference.AnnotationReference
+import com.squareup.anvil.compiler.scope
+import org.jetbrains.kotlin.backend.common.extensions.IrPluginContext
+import org.jetbrains.kotlin.ir.expressions.IrConstructorCall
+import org.jetbrains.kotlin.ir.util.parentAsClass
+import org.jetbrains.kotlin.name.FqName
+import kotlin.LazyThreadSafetyMode.NONE
+
+internal class AnnotationReferenceIr(
+  val annotation: IrConstructorCall,
+  val classReference: ClassReferenceIr,
+  private val declaringClass: ClassReferenceIr?
+) {
+  val fqName: FqName
+    get() = classReference.fqName
+
+  private val context: IrPluginContext
+    get() = classReference.context
+
+  val scope: ClassReferenceIr by lazy(NONE) {
```

Comment:
Nit: this doesn't need to be lazy. A getter is enough.

Reply:
In retrospect, lazy also feels weird on this one compared to the nullable one